### ISSUE TITLE
PY3: Cannot encrypt from a file

### ISSUE THREAD

**Issue by doktorstick** [isislovecruft/python-gnupg#94]
I haven't figured out how to use the ``encrypt`` interface to stream from a file. I've tried passing the filename and the handle from ``open``, both with no avail. The docs suggest one should be possible ("``:param str data: The file or bytestream to encrypt.``").

```python
import gnupg
import os

print ('gnupg version:', gnupg.__version__)

infile = os.path.expanduser ('~/usr/tmp/ts.txt')
with open (infile, 'rb') as f:
    kwargs = dict (passphrase='speedtest',
                   symmetric=True,
                   cipher_algo='AES256',
                   armor=False,
                   encrypt=False,
                   output='/tmp/purernd.enc.gnupg')

    gpg = gnupg.GPG (binary='/usr/bin/gpg')
    r = gpg.encrypt (f, None, **kwargs)
    print (r.ok)
```
```
gnupg version: 1.4.0
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-25-5db62b4adb3f> in <module>()
     14 
     15     gpg = gnupg.GPG (binary='/usr/bin/gpg')
---> 16     r = gpg.encrypt (f, None, **kwargs)
     17     print (r.ok)
     18 

/mnt/hgfs/work/branches/jayne/virts/jayne/lib/python3.4/site-packages/gnupg/gnupg.py in encrypt(self, data, *recipients, **kwargs)
    953         .. seealso:: :meth:`._encrypt`
    954         """
--> 955         stream = _make_binary_stream(data, self._encoding)
    956         result = self._encrypt(stream, recipients, **kwargs)
    957         stream.close()

/mnt/hgfs/work/branches/jayne/virts/jayne/lib/python3.4/site-packages/gnupg/_util.py in _make_binary_stream(s, encoding)
    396                 s = s.encode(encoding)
    397         from io import BytesIO
--> 398         rv = BytesIO(s)
    399     except ImportError:
    400         rv = StringIO(s)

TypeError: '_io.BufferedReader' does not support the buffer interface
```

**Comment by isislovecruft** [comment]
This should be fixed in 2.0.1 (because #90 and #92 were merged).

**Comment by isislovecruft** [comment]
Ignore my last comment, #92 didn't fix the issue, nor did my Python3 fixes for #92.

Additionally, when running @doktorstick's example code with Python2, I get the following traceback:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-92da3f610976> in <module>()
----> 1 r = gpg.encrypt(f, None, **kwargs)

/home/isis/code/riseup/python-gnupg/gnupg/gnupg.py in encrypt(self, data, *recipients, **kwargs)
    970             stream = data
    971         else:
--> 972             stream = _make_binary_stream(data, self._encoding)
    973         result = self._encrypt(stream, recipients, **kwargs)
    974         stream.close()

/home/isis/code/riseup/python-gnupg/gnupg/_util.py in _make_binary_stream(s, encoding)
    466         else:
    467             if type(s) is not str:
--> 468                 s = s.encode(encoding)
    469         from io import BytesIO
    470         rv = BytesIO(s)

AttributeError: 'file' object has no attribute 'encode'
```

**Comment by isislovecruft** [comment]
Oops, I accidentally wrote "fixes #93" in the commit message for 0c87da3d7, but that commit actually fixes this bug.

**Comment by isislovecruft** [comment]
This is merged and will be in the python-gnupg-2.0.1 release.
